src/custom/hooks/Trades/TradesMod.ts

```diff
@@ -0,0 +1,180 @@
+//     [allPairs]
+//   )
+// }
+
+// const MAX_HOPS = 3
+
+// /**
+//  * Returns the best trade for the exact amount of tokens in to the given token out
+//  */
+// export function useTradeExactIn(currencyAmountIn?: CurrencyAmount, currencyOut?: Currency): Trade | null {
+//   const allowedPairs = useAllCommonPairs(currencyAmountIn?.currency, currencyOut)
+
+//   const [singleHopOnly] = useUserSingleHopOnly()
+
+//   return useMemo(() => {
+//     if (currencyAmountIn && currencyOut && allowedPairs.length > 0) {
+//       if (singleHopOnly) {
+//         return (
+//           Trade.bestTradeExactIn(allowedPairs, currencyAmountIn, currencyOut, { maxHops: 1, maxNumResults: 1 })[0] ??
+//           null
+//         )
+//       }
+//       // search through trades with varying hops, find best trade out of them
+//       let bestTradeSoFar: Trade | null = null
+//       for (let i = 1; i <= MAX_HOPS; i++) {
+//         const currentTrade: Trade | null =
+//           Trade.bestTradeExactIn(allowedPairs, currencyAmountIn, currencyOut, { maxHops: i, maxNumResults: 1 })[0] ??
+//           null
+//         // if current trade is best yet, save it
+//         if (isTradeBetter(bestTradeSoFar, currentTrade, BETTER_TRADE_LESS_HOPS_THRESHOLD)) {
+//           bestTradeSoFar = currentTrade
+//         }
+//       }
+//       return bestTradeSoFar
+//     }
+
+//     return null
+//   }, [allowedPairs, currencyAmountIn, currencyOut, singleHopOnly])
+// }
+
+// /**
+//  * Returns the best trade for the token in to the exact amount of token out
+//  */
+// export function useTradeExactOut(currencyIn?: Currency, currencyAmountOut?: CurrencyAmount): Trade | null {
+//   const allowedPairs = useAllCommonPairs(currencyIn, currencyAmountOut?.currency)
+
+//   const [singleHopOnly] = useUserSingleHopOnly()
+
+//   return useMemo(() => {
+//     if (currencyIn && currencyAmountOut && allowedPairs.length > 0) {
+//       if (singleHopOnly) {
+//         return (
+//           Trade.bestTradeExactOut(allowedPairs, currencyIn, currencyAmountOut, { maxHops: 1, maxNumResults: 1 })[0] ??
+//           null
+//         )
+//       }
+//       // search through trades with varying hops, find best trade out of them
+//       let bestTradeSoFar: Trade | null = null
+//       for (let i = 1; i <= MAX_HOPS; i++) {
+//         const currentTrade =
+//           Trade.bestTradeExactOut(allowedPairs, currencyIn, currencyAmountOut, { maxHops: i, maxNumResults: 1 })[0] ??
+//           null
+//         if (isTradeBetter(bestTradeSoFar, currentTrade, BETTER_TRADE_LESS_HOPS_THRESHOLD)) {
+//           bestTradeSoFar = currentTrade
+//         }
+//       }
+//       return bestTradeSoFar
+//     }
+//     return null
+//   }, [currencyIn, currencyAmountOut, allowedPairs, singleHopOnly])
+// }
+
+export function useIsTransactionUnsupported(currencyIn?: Currency, currencyOut?: Currency): boolean {
+  const unsupportedTokens: { [address: string]: Token } = useUnsupportedTokens()
```

Comment:
There was this hook before. I would suggest as I mentioned above to make a useIsTokenSupported() that returns a predicate function.

This hook internally could use `useUnsupportedTokens()` and our own list of unsupported tokens